opm/simulators/linalg/ISTLSolverEbos.hpp

```diff
@@ -180,34 +141,33 @@ DenseMatrix transposeDenseMatrix(const DenseMatrix& M)
             }
 #endif
             extractParallelGridInformationToISTL(simulator_.vanguard().grid(), parallelInformation_);
+
+            // For some reason simulator_.model().elementMapper() is not initialized at this stage
+            // Hence const auto& elemMapper = simulator_.model().elementMapper(); does not work.
+            // Set it up manually
+            using ElementMapper =
+                Dune::MultipleCodimMultipleGeomTypeMapper<GridView>;
+            ElementMapper elemMapper(simulator_.vanguard().grid().leafGridView(), Dune::mcmgElementLayout());
+            detail::findOverlapAndInterior(simulator_.vanguard().grid(), elemMapper, overlapRows_, interiorRows_);
+
             useWellConn_ = EWOMS_GET_PARAM(TypeTag, bool, MatrixAddWellContributions);
-            ownersFirst_ = EWOMS_GET_PARAM(TypeTag, bool, OwnerCellsFirst);
-            interiorCellNum_ = detail::numMatrixRowsToUseInSolver(simulator_.vanguard().grid(), ownersFirst_);
-
-            if ( isParallel() && (!ownersFirst_ || parameters_.linear_solver_use_amg_) ) {
-                detail::setWellConnections(gridForConn, simulator_.vanguard().schedule().getWellsatEnd(), useWellConn_, wellConnectionsGraph_);
-                // For some reason simulator_.model().elementMapper() is not initialized at this stage
-                // Hence const auto& elemMapper = simulator_.model().elementMapper(); does not work.
-                // Set it up manually
-                using ElementMapper =
-                    Dune::MultipleCodimMultipleGeomTypeMapper<GridView>;
-                ElementMapper elemMapper(simulator_.vanguard().gridView(), Dune::mcmgElementLayout());
-                detail::findOverlapAndInterior(gridForConn, elemMapper, overlapRows_, interiorRows_);
-                noGhostAdjacency();
-                setGhostsInNoGhost(*noGhostMat_);
-                if (ownersFirst_)
-                    OpmLog::warning("OwnerCellsFirst option is true, but ignored.");
+            const bool ownersFirst = EWOMS_GET_PARAM(TypeTag, bool, OwnerCellsFirst);
+            if (!ownersFirst) {
+                const std::string msg = "The linear solver no longer supports --owner-cells-first=false.";
+                if (on_io_rank) {
+                    OpmLog::error(msg);
+                }
+                OPM_THROW_NOLOG(std::runtime_error, msg);
             }
 
-            if (useFlexible_)
-            {
-                // Print parameters to PRT/DBG logs.
-                if (simulator.gridView().comm().rank() == 0) {
-                    std::ostringstream os;
-                    os << "Property tree for linear solver:\n";
-                    boost::property_tree::write_json(os, prm_, true);
-                    OpmLog::note(os.str());
-                }
+            interiorCellNum_ = detail::numMatrixRowsToUseInSolver(simulator_.vanguard().grid(), true);
```

Comment:
This is now actually doing `grid.numCells()`

Reply:
I do not think so. This is the function called, with the second argument being true:
```
    template <class Grid>
    size_t numMatrixRowsToUseInSolver(const Grid& grid, bool ownerFirst)
    {
        size_t numInterior = 0;
        if (!ownerFirst || grid.comm().size()==1)
            return grid.numCells();
        const auto& gridView = grid.leafGridView();
        auto elemIt = gridView.template begin<0>();
        const auto& elemEndIt = gridView.template end<0>();

        // loop over cells in mesh
        for (; elemIt != elemEndIt; ++elemIt) {

            // Count only the interior cells.
            if (elemIt->partitionType() == Dune::InteriorEntity) {
                numInterior++;
            }
        }

        return numInterior;
    }
```
Looks fine to me? Could be simplified if we go with this though, as we would never call it with a `false` argument though.